### ISSUE TITLE
Replace housing queue calculation with input list

### ISSUE THREAD

**Issue by mbillow** [ComputerScienceHouse/conditional#103]
The housing queue would be easier as a list input by the evals director. This was addressed as a part of the discussion we had during Evals about Conditional.

**Comment by mbillow** [comment]
As I am working on #89, I am not going to spend the time to convert all of the logic over to the new library.

**Comment by stevenmirabito** [comment]
Just to confirm as I wasn't present for the discussion, there is definitely no way to do this correctly in our business logic? IIRC there was some incompatibility with our process and doing it automatically, but I just want to make sure. I completely understand the want for this since it's not working properly now, but obviously I would prefer the automated approach if it's possible. If not, no big deal, this wouldn't be too difficult to implement, although it would require a schema addition.

**Comment by mbillow** [comment]
Meghan made the point that the ability to be in the housing queue does not automatically add you to it. We can calculate who is eligible to be in the list, but the evaluations director should have control over the members in the list.

**Comment by stevenmirabito** [comment]
Oh, that's easy then, we just need to make sure our eligibility criteria is correct. For the UI I'm thinking something like this:

![](http://i.imgur.com/w8RiwVO.png)

The evals director would be able to hide people from the queue who are eligible but are not seeking housing.

**Comment by mbillow** [comment]
I want to be carefull of our interface getting in the way of the process instead of helping it. I would much rather see a list that can be populated by the evaluations director where we handle things like sorting, etc. once we get that list.

**Comment by stevenmirabito** [comment]
I mean, the only case where that would be a problem is if the evaluations director wants to put someone in the queue that isn't eligible to be in the queue. Even if we wanted that, it can be done fairly easily with an "add to queue" control. I think automatically populating and removing people from the queue is valuable and definitely helps the process.

I guess I'm just not seeing the case where manually defining the list is anything but more work for the evaluations director, assuming we correct our eligibility logic.

**Comment by mbillow** [comment]
The correctness of our logic is dependent on proper input and tracking of co-ops... (#80). That is most likely going to include schema and logic changes that will cause conflicts with this if developed in parallel. The functionality we are losing was ignored because it was always incorrect. We can discuss how we want to implement the new housing queue here, but it shouldn't block #89.

**Comment by stevenmirabito** [comment]
This is blocking #89 because the existing functionality was ripped out in preparation for this. Unless the functionality is restored and #89 no longer regresses on that feature, this blocks it.

**Comment by mbillow** [comment]
I'm not sure what you mean by consequences. The only other page that uses the blueprint that uses the logic removed in #89 is the dashboard. We aren't going to cut a version and merge into master until both #89 and this issue are resolved (along with everything else tagged for v1.3). We aren't regressing, we are removing flawed functionality with the intent of building something better in its place. #89 is the first step in that direction.

**Comment by mbillow** [comment]
As per conversation in #conditional on Slack, the scope we are going to populate the list is (CurrentStudents - MembersWithoutOnFloor - MembersWithRm#) for the evaluations director. Then once a user has expressed interest in living on floor we can add them to some new model and only display those who have expressed interest on the front end (using @stevenmirabito's interface above).
